frontend-react/src/hooks/UseLookupTable.ts

```diff
@@ -0,0 +1,106 @@
+import { useEffect, useState } from "react";
+import axios from "axios";
+
+import {
+    lookupTableApi,
+    LookupTable,
+    ValueSet,
+    LookupTables,
+} from "../network/api/LookupTableApi";
+import { showError } from "../components/AlertNotifications";
+
+export const generateUseLookupTable =
+    <T>(tableName: LookupTables) =>
+    () => {
+        return useLookupTable<T>(tableName);
+    };
+
+export async function getLatestVersion(
+    tableName: LookupTables
+): Promise<number> {
+    let response;
+    try {
+        response = await axios(lookupTableApi.getTableList()).then(
+            (response) => response.data
+        );
+
+        let filteredBody: LookupTable[] = response.filter(
+            (tv: LookupTable) => tv.tableName === tableName && tv.isActive
+        );
+        if (filteredBody.length === 0) {
+            filteredBody = response.filter(
+                (tv: LookupTable) => tv.tableName === tableName
+            );
+        }
+        const table: LookupTable = filteredBody.sort(
+            (a: LookupTable, b: LookupTable) =>
+                b["tableVersion"] - a["tableVersion"]
+        )[0];
+
+        return table?.tableVersion;
+    } catch (e: any) {
+        console.trace(e);
+        showError(
+            `An error occurred while retrieving the latest version: ${e.toString()}`
+        );
+        return -1;
+    }
+}
+
+export async function getLatestData<T>(
+    version: number,
+    tableName: string
+): Promise<T | T[]> {
+    const endpointHeader = lookupTableApi.getTableData<T>(version, tableName);
+
+    try {
+        return await axios
+            .get<T>(endpointHeader.url, endpointHeader)
+            .then((response) => response.data);
+    } catch (e: any) {
+        console.trace(e);
+        showError(
+            `An error occurred while retrieving the latest data: ${e.toString()}`
+        );
+        return new Promise<T[] | T>(() => Promise.resolve());
```

Comment:
I don't think you'd have a problem just returning `[]` here. Async functions should be smart enough to cast that into a promise.

Reply:
@doug-s-nava ha! 🤦‍♂️ you're right, and I tried everything _but_ that